apps/playlist/apis/playlist/v0alpha1/playlist_object_gen.go

```diff
@@ -0,0 +1,265 @@
+//
+// Code generated by grafana-app-sdk. DO NOT EDIT.
+//
+
+package v0alpha1
+
+import (
+	"fmt"
+	"github.com/grafana/grafana-app-sdk/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"time"
+)
+
+// +k8s:openapi-gen=true
+type Playlist struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              PlaylistSpec   `json:"spec"`
+	PlaylistStatus    PlaylistStatus `json:"status"`
+}
+
+func (o *Playlist) GetSpec() any {
+	return o.Spec
+}
+
+func (o *Playlist) SetSpec(spec any) error {
+	cast, ok := spec.(PlaylistSpec)
+	if !ok {
+		return fmt.Errorf("cannot set spec type %#v, not of type Spec", spec)
+	}
+	o.Spec = cast
+	return nil
+}
+
+func (o *Playlist) GetSubresources() map[string]any {
```

Comment:
How does this relate to subresources that are dynamic eg "pod/logs" or for us things like "/access" or "/history"

I may be missing why/where this is useful abstraction since IIUC, it is always the named properties on the object that exists

Reply:
I don't believe it's used anymore at this point, but it was used prior to the `v0.15.x` app-sdk as part of the kubernetes abstraction. We can look into dropping these methods entirely once we verify that they are not being depended upon by any existing apps.